src/shadow.rs

```diff
@@ -0,0 +1,174 @@
+//! Manipulate the contents of the shadow password file, `/etc/shadow`.
+use std::ffi::CStr;
+use std::ffi::CString;
+
+/// Represents an entry in `/etc/shadow`.
+// Documentation is based on the `shadow(5)` and `shadow(3)` man pages.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct Shadow {
+    /// User name
+    pub name: String,
+    /// Encrypted password
+    ///
+    /// Refer to crypt(3) for details on how this string is interpreted.
+    /// If the password field contains some string that is not a valid
+    /// result of crypt(3), for instance ! or *, the user will not be able to
+    /// use a unix password to log in (but the user may log in the system by
+    /// other means).
+    ///
+    /// This field may be empty, in which case no passwords are required to
+    /// authenticate as the specified login name. However, some applications
+    /// which read the /etc/shadow file may decide not to permit any access at
+    /// all if the password field is empty.
+    ///
+    /// A password field which starts with a exclamation mark means that the
+    /// password is locked.
+    pub password: CString,
```

Comment:
Why use `String` for name, but `CString` for password?

Reply:
Because of what I read from @ctrlcctrlv here: https://github.com/nix-rust/nix/pull/864/files
> Because it's guaranteed that the fields I made `String` will conform to [`NAME_REGEX`](https://serverfault.com/a/73101/407341). `gecos` ("real name") has no such guarantee.